contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,952 @@
+
+        uint160 currentPrice = getPoolX96Price();
+        // check we are in inspected tick range
+        require(
+            currentPrice >= sqrtRatioX96Tick0 &&
+                currentPrice <= sqrtRatioX96Tick1,
+            "Not in expected tick range"
+        );
+
+        uint256 currentWethShare = 0;
+        if (wethPositionBalance != 0) {
+            currentWethShare = wethPositionBalance.divPrecisely(
+                wethPositionBalance + oethbPositionBalance
+            );
+        }
+
+        uint256 wethDiff = Math.max(poolWethShare, currentWethShare) -
+            Math.min(poolWethShare, currentWethShare);
+
+        if (wethDiff < poolWethShareVarianceAllowed) {
+            emit PoolRebalanced(
+                currentWethShare,
+                poolWethShare,
+                wethPositionBalance,
+                oethbPositionBalance
+            );
+        } else {
+            revert PoolRebalanceOutOfBounds(
+                currentWethShare,
+                poolWethShare,
+                wethPositionBalance,
+                oethbPositionBalance
+            );
+        }
+    }
+
+    /**
+     * Burns any lingering OETHb tokens still remaining on the contract
+     */
+    function _burnOethbOnTheContract() internal {
+        uint256 oethbBalance = IERC20(OETHb).balanceOf(address(this));
+        IVault(vaultAddress).burnForStrategy(oethbBalance);
+    }
+
+    function _updateUnderlyingAssets() internal {
+        if (tokenId == 0) {
+            underlyingAssets = 0;
+        } else {
+            uint128 liquidity = _getLiquidity();
+
+            /**
+             * Our net value represent the smallest amount of tokens we are able to extract from the position
+             * given our liquidity.
+             *
+             * The least amount of tokens extraditable from the position is where the active trading price is
+             * at the ticker 0 meaning the pool is offering 1:1 trades between WETH & OETHb. At that moment the pool
+             * consists completely of OETHb and no WETH.
+             *
+             * The more swaps from WETH -> OETHb happen on the pool the more the price starts to move towards the -1
+             * ticker making OETHb (priced in WETH) more expensive.
+             */
+            (uint256 wethAmount, uint256 oethbAmount) = helper
+                .getAmountsForLiquidity(
+                    sqrtRatioX96TickClosestToParity, // sqrtRatioX96
+                    sqrtRatioX96Tick0, // sqrtRatioAX96
+                    sqrtRatioX96Tick1, // sqrtRatioBX96
+                    liquidity
+                );
+
+            require(wethAmount == 0, "Non zero wethAmount");
+            underlyingAssets = oethbAmount;
+        }
+    }
+
+    /**
+     * @notice Deposit all supported assets in this strategy contract to the platform
+     */
+    function depositAll() external override onlyVault nonReentrant {
+        _deposit(WETH, IERC20(WETH).balanceOf(address(this)));
```

Comment:
nit: ``balance > 0` check would be great to have here

Reply:
agreed: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/21fc9e8b649d0c2437d3fb4a603291dea4154d18